calico_containers/tests/st/utils/workload.py

```diff
@@ -79,12 +79,16 @@ def _get_ping_function(self, ip):
         The function raises a CommandExecError exception if the ping fails,
         or returns the output of the ping.
         """
-        version = IPAddress(ip).version
-        assert version in [4, 6]
-        if version == 4:
-            ping = "ping"
-        else:  # if version == 6:
-            ping = "ping6"
+        # Default to "ping"
+        ping = "ping"
+
+        try:
+            version = IPAddress(ip).version
+            assert version in [4, 6]
+            if version == 6:
+                ping = "ping6"
```

Comment:
`ping = "ping" if version ==4 else "ping6"` might be nicer than defining it and overwriting it